diff --git a/arch/arm64/boot/dts/xilinx/Makefile b/arch/arm64/boot/dts/xilinx/Makefile
index 1517d8c9849..f6d3f5e3cfc 100644
--- a/arch/arm64/boot/dts/xilinx/Makefile
+++ b/arch/arm64/boot/dts/xilinx/Makefile
@@ -1,5 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0
 dtb-$(CONFIG_ARCH_ZYNQMP) += avnet-ultra96-rev1.dtb
+dtb-$(CONFIG_ARCH_ZYNQMP) += avnet-ultra96-v2-rev1.dtb
 dtb-$(CONFIG_ARCH_ZYNQMP) += zynqmp-zc1232-revA.dtb
 dtb-$(CONFIG_ARCH_ZYNQMP) += zynqmp-zc1254-revA.dtb
 dtb-$(CONFIG_ARCH_ZYNQMP) += zynqmp-zc1751-xm015-dc1.dtb
diff --git a/arch/arm64/boot/dts/xilinx/avnet-ultra96-v2-rev1.dts b/arch/arm64/boot/dts/xilinx/avnet-ultra96-v2-rev1.dts
new file mode 100644
index 00000000000..31b78ab7fde
--- /dev/null
+++ b/arch/arm64/boot/dts/xilinx/avnet-ultra96-v2-rev1.dts
@@ -0,0 +1,47 @@
+/dts-v1/;
+
+#include "zynqmp-zcu100-revC.dts"
+
+/ {
+	model = "Avnet Ultra96-V2 Rev1";
+	compatible = "avnet,ultra96-v2-rev1", "avnet,ultra96-v2", "xlnx,zynqmp";
+};
+
+&sdio_pwrseq {
+	chip_en-gpios = <&gpio 8 GPIO_ACTIVE_LOW>;
+};
+
+&uart0 {
+	/delete-node/ bluetooth;
+};
+
+&gpio {
+	/delete-property/ gpio-line-names;
+};
+
+&sdhci1 {
+	max-frequency = <50000000>;
+	/delete-property/ cap-power-off-card;
+	wilc_sdio@1 {
+		compatible = "microchip,wilc3000";
+		reg = <0>;
+		bus-width = <0x4>;
+		status = "okay";
+	};
+	/delete-node/ wifi@2;
+};
+
+&i2csw_4 {
+	irps5401_43: irps5401@43 { /* U11 on Ultra96-V2 */
+		reg = <0x43>;
+		compatible = "infineon,irps5401";
+	};
+	irps5401_44: irps5401@44 { /* U12 on Ultra96-V2 */
+		reg = <0x44>;
+		compatible = "infineon,irps5401";
+	};
+	ir38060_45: ir38060@45 { /* U21 on Ultra96-V2 */
+		reg = <0x45>;
+		compatible = "infineon,ir38060";
+	};
+};
